stdlib/Profile/src/Profile.jl

```diff
@@ -182,8 +198,25 @@ a dictionary `lidict` of line information.
 
 See `Profile.print([io], data)` for an explanation of the valid keyword arguments.
 """
-print(data::Vector{<:Unsigned} = fetch(), lidict::Union{LineInfoDict, LineInfoFlatDict} = getdict(data); kwargs...) =
-    print(stdout, data, lidict; kwargs...)
+function print(io::IO = stdout, data::Vector{<:Unsigned} = fetch(), lidict::Union{LineInfoDict, LineInfoFlatDict} = getdict(data); separate_threads = true, kwargs...)
+    if separate_threads
+        kwargs = filter(p -> !in(first(p), [:threads, :separate_threads]),  kwargs)
+        println(io, "Overhead ╎ [+additional indent] Count File:Line; Function")
+        println(io, "=========================================================")
+        for tid in 1:Threads.nthreads()
+            taskids = get_task_ids(data, tid)
+            printstyled(io, "Thread $tid\n"; bold=true, color=Base.info_color())
+            for taskid in taskids
+                printstyled(io, "Task $(Base.repr(taskid))\n"; bold=true, color=Base.debug_color())
+                _print(io, data, lidict; threads = tid:tid, tasks = [taskid], header = false, kwargs...)
+                println(io)
+            end
+        end
+    else
+        kwargs = filter(p -> first(p) != :separate_threads,  kwargs)
+        _print(io, data, lidict; kwargs...)
+    end
```

Comment:
Even if you don't separate the threads, stack traces will still be _grouping by_ the threads, since it will be present in their stack trace! So we might want to consider skipping/removing the thread and task ids if the user doesn't want them? Kind of like `C=false`?